src/tally.F90

```diff
@@ -3138,6 +3149,128 @@ subroutine score_collision_tally(p)
+
+  !end if
+    integer :: i
+    integer :: i_tally
+    integer :: i_filt
+    integer :: i_bin
+    integer :: j                    ! loop index for scoring bins
+    integer :: filter_index         ! single index for single bin
+    integer :: matching_bin         ! next valid filter bin
+    real(8) :: flux                 ! collision estimate of flux
+    real(8) :: filter_weight        ! combined weight of all filters
+    logical :: finished             ! found all valid bin combinations
+    type(TallyObject), pointer :: t
+
+    ! Determine collision estimate of flux
+    if (survival_biasing) then
+      ! We need to account for the fact that some weight was already absorbed
+      flux = (p % last_wgt + p % absorb_wgt)
+    else
+      flux = p % last_wgt
+    end if
+
+    TALLY_LOOP: do i = 1, active_cell_to_cell_tallies % size()
+      ! Get index of tally and pointer to tally
+      i_tally = active_cell_to_cell_tallies % data(i)
+      t => tallies(i_tally)
+
+      ! Find all valid bins in each filter if they have not already been found
+      ! for a previous tally.
+      do j = 1, size(t % filter)
+        i_filt = t % filter(j)
+        if (.not. filter_matches(i_filt) % bins_present) then
+          call filter_matches(i_filt) % bins % clear()
+          call filter_matches(i_filt) % weights % clear()
+          matching_bin = NO_BIN_FOUND
+          do
+            call filters(i_filt) % obj % get_next_bin(p, t % estimator, &
+                 matching_bin, matching_bin, filter_weight)
+            if (matching_bin == NO_BIN_FOUND) exit
+            call filter_matches(i_filt) % bins % push_back(matching_bin)
+            call filter_matches(i_filt) % weights % push_back(filter_weight)
+          end do
+          filter_matches(i_filt) % bins_present = .true.
+        end if
+        ! If there are no valid bins for this filter, then there is nothing to
+        ! score and we can move on to the next tally.
+        if (filter_matches(i_filt) % bins % size() == 0) cycle TALLY_LOOP
+
+        ! Set the index of the bin used in the first filter combination
+        filter_matches(i_filt) % i_bin = 1
+      end do
+
+      ! ========================================================================
+      ! Loop until we've covered all valid bins on each of the filters.
+
+      FILTER_LOOP: do
+
+        ! Reset scoring index and weight
+        filter_index = 1
+        filter_weight = ONE
+
+        ! Determine scoring index and weight for this filter combination
+        do j = 1, size(t % filter)
+          i_filt = t % filter(j)
+          i_bin = filter_matches(i_filt) % i_bin
+          filter_index = filter_index + (filter_matches(i_filt) % bins % &
+               data(i_bin) - 1) * t % stride(j)
+          filter_weight = filter_weight * filter_matches(i_filt) % weights % &
+               data(i_bin)
+        end do
+
+        call score_general(p, t, 0, filter_index, &
+             0, 0d0, flux * filter_weight)
```

Comment:
It doesn't make sense for this function to call `score_general`.  There would be no reason to have say a `scatter` score that is only called when a particle crosses a surface.  Just put the relevant code from `score_general` directly in this function.